src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala

```diff
@@ -736,6 +738,8 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
 
         if (m.isConstructor) {
           val specCtor = enterMember(cloneInSpecializedClass(m, x => x))
+          if (settings.unitSpecialization && specCtor.paramLists.exists(_.exists(_.typeSignature =:= definitions.UnitTpe)))
+            reporter.warning(m.pos, "Constructor is specialized for a Unit-valued parameter")
```

Comment:
let's offer the user what they should do:

```suggestion
            reporter.warning(m.pos, "fruitless specialization of Unit in the parameter position; specify specific types as `@specialized(Int, Double, Boolean)` instead")
```

Reply:
`fruitless` is one of my favorite lint words; I wish I had thought of it. Also `bootless`.